tenders/spt/spt_core.c

```diff
@@ -146,6 +146,28 @@ struct spt *spt_init(size_t mem_size)
     return spt;
 }
 
+int spt_guest_mprotect(void *t, uint64_t addr_start, uint64_t addr_end,
+        int prot)
+{
+    struct spt *spt = t;
+
+    assert(addr_start <= spt->mem_size);
+    assert(addr_end <= spt->mem_size);
+    assert(addr_start < addr_end);
+
+    uint8_t *vaddr_start = spt->mem + addr_start;
+    assert(vaddr_start >= spt->mem);
+    size_t size = addr_end - addr_start;
+    assert(size > 0 && size <= spt->mem_size);
+
+    /*
+     * On spt, there is no distinction between host-side and guest-side memory
+     * protection, so just pass through to mprotect() directly, which will do
+     * the right thing.
+     */
+    return mprotect(vaddr_start, size, prot);
```

Comment:
Do we have use-cases for `W && X` mappings, or would it make sense to prevent the creation of such mappings?

Reply:
> Do we have use-cases for W && X mappings, or would it make sense to prevent the creation of such mappings?

Already done by the caller (ELF loader) at https://github.com/Solo5/solo5/blob/master/tenders/common/elf.c#L344, I don't want to do that check here.

Reply:
Ah, I had missed that that took place just above the call site in `elf_load`.